code/game/objects/items/weapons/tanks/tanks.dm

```diff
@@ -117,17 +117,18 @@
 	user.visible_message("<span class='suicide'>[user] is putting [src]'s valve to [user.p_their()] lips! It looks like [user.p_theyre()] trying to commit suicide!</span>")
 	playsound(loc, 'sound/effects/spray.ogg', 10, 1, -3)
 	if (H && !qdeleted(H))
-		for(var/obj/item/W in H)
-			H.dropItemToGround(W)
-			if(prob(50))
-				step(W, pick(alldirs))
-		H.hair_style = "Bald"
-		H.update_hair()
-		H.bleed_rate = 5
-		new /obj/effect/gibspawner/generic(H.loc, H.viruses, H.dna)
-		H.adjustBruteLoss(1000) //to make the body super-bloody
-
-	return (BRUTELOSS)
+		spawn(0)
+			var/list/items = H.get_equipped_items()
+			for(var/obj/item/i in items)
+				H.dropItemToGround(i)
+			H.drop_all_held_items()
+			for(var/obj/item/i in get_turf(H))
+				i.add_blood(H)
+			if(H.loc)
+				src.loc = null
+				src.loc = H.loc
```

Comment:
This part of the code is to move the tank from the bottom of the stack of items they drop to the top.  I don't know of any other way to move the item to the top of the stuff pile without making the location null then setting it back. Moving the tank to the top so it makes it easier to determine if someone dies by oxygen tank suicide as all their items they drop and the tank will have blood and the tank will be on the top.